### PR TITLE
[Backend bug fixed] duplicate client and customer and error handled

### PR DESCRIPTION
**Pull request by OvilashMoitra**
I have added logic where the user will not able to create customer and employee with same email and phone number. Also handled the error on the controller

### CODE REVIEW

**Comment by OvilashMoitra** [comment]
https://www.loom.com/share/8208a4244b504543a601b95e7d2801b5?sid=6dd32684-70cd-47c8-93fd-9f493c96115b
@polymahh can you please review it. I have added constraint on creating as well as updating the document.

**Comment by polymahh** [comment]
thank you @OvilashMoitra for your help, i can see from the video that it is working correctly only showing an err if all fields are the same.
this is a backend issue, so i'll leave it for the backend devs to review it and test it 
@salahlalami @adeojoemmanuel

**Comment by OvilashMoitra** [comment]
@polymahh Thank you. I think it's because the message I have sent from the backend.I can also change the message if the message is not perfect.

File: handlers/isClientExist.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const isClientExist = async (Model, req) => {
+    // check for existing employee
+    const { company, managerSurname, managerName, email, phone } = req.body
+    let isExist;
+    if (company || managerSurname || managerName || email || phone) {
+        isExist = await Model.findOne({ company, managerName, managerName, email, phone })
```
Comment (Ando22) [inline handlers/isClientExist.js:6]: I see you mention `managerName` twice. please change this 🙏

**Review by Ando22 — CHANGES_REQUESTED** [review]

File: handlers/isClientExist.js (hunk 1 above)
Comment (OvilashMoitra) [reply handlers/isClientExist.js:6]: My bad. Let me change it.

**Review by Ando22 — APPROVED** [review]

**Comment by OvilashMoitra** [comment]
@salahlalami can you review it.

File: handlers/isClientExist.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const isClientExist = async (Model, req) => {
+    // check for existing employee
+    const { company, managerSurname, managerName, email, phone } = req.body
+    let isExist;
+    if (company || managerSurname || managerName || email || phone) {
+        isExist = await Model.findOne({ company, managerName, managerSurname, email, phone })
```
Comment (salahlalami) [inline handlers/isClientExist.js:6]: please verify only email & phone

**Review by salahlalami — CHANGES_REQUESTED** [review]

File: handlers/isClientExist.js (hunk 2 above)
Comment (OvilashMoitra) [reply handlers/isClientExist.js]: image
@salahlalami can you check now

Comment (salahlalami) [reply handlers/isClientExist.js]: @OvilashMoitra remove unused variable , keep only phone & email

Comment (salahlalami) [reply handlers/isClientExist.js]: @OvilashMoitra check also if company name exist ,
if exist end  return id of client

Comment (OvilashMoitra) [reply handlers/isClientExist.js]: image
image
@salahlalami Can you check now.

**Review by salahlalami — CHANGES_REQUESTED** [review]
please update & pull from dev , and try to start again , don't touch coreController

**Comment by salahlalami** [comment]
Thank you @OvilashMoitra we need to close this pr , is outdated

**Pull request closed without merging**